React/React Pong/src/components/Board.tsx

```diff
@@ -0,0 +1,187 @@
+import { useEffect, useState } from "react";
+import Ball from "./Ball";
+import Paddle from "./Paddle";
+import "./board.css";
+import { CONTROLS, DIRECTIONS } from "../enums";
+
+export default function Board() {
+    const [leftPaddle, setLeftPaddle] = useState(8);
+    const [rightPaddle, setRightPaddle] = useState(8);
+
+    const initialBallState = { x: 10, y: 15 };
+    const [ball, setBall] = useState(initialBallState);
+    const [velocity, setVelocity] = useState({ x: 1, y: 1 });
+    const [speed, setSpeed] = useState(140);
+    const [direction, setDirection] = useState(DIRECTIONS.RIGHT);
+    const [wallHit, setWallHit] = useState(false);
+
+    const [playerOneScore, setPlayerOneScore] = useState(0);
+    const [playerTwoScore, setPlayerTwoScore] = useState(0);
+
+    const [gameStarted, setGameStarted] = useState(false);
+
+    useEffect(() => {
+        function handleKeyDown(event: KeyboardEvent) {
+            switch (event.key) {
+                case CONTROLS.UP:
+                    if (rightPaddle >= 1) {
+                        setRightPaddle(rightPaddle - 1);
+                        return;
+                    }
+                    break;
+                case CONTROLS.W:
+                    if (leftPaddle >= 1) {
+                        setLeftPaddle(leftPaddle - 1);
+                        return;
+                    }
+                    break;
+                case CONTROLS.DOWN:
+                    if (rightPaddle <= 15) {
+                        setRightPaddle(rightPaddle + 1);
+                        return;
+                    }
+                    break;
+                case CONTROLS.S:
+                    if (leftPaddle <= 15) {
+                        setLeftPaddle(leftPaddle + 1);
+                        return;
+                    }
+                    break;
+                case CONTROLS.SPACE:
+                    setGameStarted(true);
+                    break;
+            }
+        }
+
+        window.addEventListener("keydown", handleKeyDown);
```

Comment:
Adding the eventListener like this will always reatach it if leftPaddle or rightPaddle are changed.
Please think of a neater solution.

Go through useEffect documentation if needed - maybe it is not the best place to attach the event listener